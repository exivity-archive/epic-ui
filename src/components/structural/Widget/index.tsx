import * as React from 'react'
import styled from 'styled-components';

const WidgetTitle = styled.h4`
  font-weight: 200;
`
const WidgetSubTitle = styled.span`
  color: #999;
  font-size: 0.8em;
  padding-left: 10px;
`

interface IWidget {
  className?: string
  children: JSX.Element | string
  title?: string
  subTitle?: string
}

const Widget: React.FC<IWidget> = ({ className, children, title, subTitle }) => (
  <div className={className}>
    {title && 
      <WidgetTitle>
        {title}
        {subTitle && <WidgetSubTitle>{subTitle}</WidgetSubTitle>}
      </WidgetTitle>}
    {children}
  </div>
)

export default styled(Widget)`
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 65%
`