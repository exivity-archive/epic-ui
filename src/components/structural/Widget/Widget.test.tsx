import * as React from 'react'

import Widget from '.'
import { shallow } from 'enzyme';

test('the widget renders with text inside', () => {
  const widget = shallow(<Widget>hello</Widget>)
  expect(widget).toMatchSnapshot()
})

test('widget renders with jsx element inside', () => {
  const widget = shallow(<Widget><div>hello</div></Widget>)
  expect(widget).toMatchSnapshot()
})

test('widget can take in a title and a subtitle', () => {
  const widget = shallow(<Widget title='Hey' subTitle="bye">hello</Widget>)
  expect(widget).toMatchSnapshot()
})