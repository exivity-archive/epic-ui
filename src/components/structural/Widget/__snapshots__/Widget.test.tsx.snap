// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the widget renders with text inside 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef>
    hello
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "hello",
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 65%
",
          ],
        },
        "displayName": "Styled(Widget)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": "hello",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "hello",
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 65%
",
            ],
          },
          "displayName": "Styled(Widget)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": "hello",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`widget can take in a title and a subtitle 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    subTitle="bye"
    title="Hey"
  >
    hello
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "hello",
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 65%
",
          ],
        },
        "displayName": "Styled(Widget)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
      "forwardedRef": null,
      "subTitle": "bye",
      "title": "Hey",
    },
    "ref": null,
    "rendered": "hello",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "hello",
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 65%
",
            ],
          },
          "displayName": "Styled(Widget)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "forwardedRef": null,
        "subTitle": "bye",
        "title": "Hey",
      },
      "ref": null,
      "rendered": "hello",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`widget renders with jsx element inside 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef>
    <div>
      hello
    </div>
  </ForwardRef>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        hello
      </div>,
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "rules": Array [
            "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 65%
",
          ],
        },
        "displayName": "Styled(Widget)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "hello",
      },
      "ref": null,
      "rendered": "hello",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          hello
        </div>,
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  background: white;
  box-shadow: 2px 2px 0 lightblue;
  padding: 20px;
  padding-bottom: 100px;
  position: relative;
  box-sizing: border-box;
  border-radius: 3px;
  width: 65%
",
            ],
          },
          "displayName": "Styled(Widget)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hello",
        },
        "ref": null,
        "rendered": "hello",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
